File: src/main/java/com/example/probation/core/dto/UserDto.java
package com.example.probation.core.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {
    private String username;
    private String description;
    private Integer rating;
    private String email;
    private Set<RoleDto> roles;
}

File: src/main/kotlin/com/example/probation/core/dto/UserDto.kt
package com.example.probation.core.dto

import com.example.probation.core.entity.Role

data class UserDto(
    var id: Long? = null,
    var username: String? = null,
    var description: String? = null,
    var rating: Int? = null,
    var email: String? = null,
    var roles: MutableSet<Role> = mutableSetOf()
)
